core/tests/protractor_utils/collectionPlayer.js

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utilities for manipulating the collection player when
+ * carrrying out end-to-end testing with protractor.
+ */
+
+var general = require('./general.js');
+var player = require('./player.js');
+
+// Visit the demo collection.
+var goToDemoCollection = function() {
+  browser.get('/collection/0');
+};
+
+// Verify that the "SIGN IN" button appears correctly in guest's view.
+var verifySignInButton = function() {
+  var text = element(by.css('.protractor-test-sign-in-button')).getText();
+  expect(text).toBe('SIGN IN');
+};
+
+// Play the first exploration in the demo collection.
+//   Assumes we are starting from the demo collection page.
+var playDemoExplorationOne = function() {
+  // Start the first exploration (from the demo collection page).
+  var explButton = element.all(by.css(
+    '.protractor-test-collection-expl-node')).get(0);
+  explButton.click();
+
+  player.submitAnswer(
+    'MultipleChoiceInput', 'It\'s translated from a different language.');
+  browser.waitForAngular();
+  player.clickThroughToNextCard();
+  
+  player.submitAnswer('TextInput', 'Finnish');
+  browser.waitForAngular();
+  general.waitForSystem(5000);
+  player.clickThroughToNextCard();
+
+  // The below may give "element not clickable" error.
+  player.submitAnswer('NumericInput', 121);
+  browser.waitForAngular();
+  player.clickThroughToNextCard();
+  // Function needs to be finished
```

Comment:
Oh, you're going to run into trouble here. The next interaction is a map and it's going to be quite hard to test that.

I'm starting to wonder whether it might be better to create a dummy collection with two or three trivial explorations instead (maybe 1 text input state and then it ends), and then play through that. The explorations in the demo collection is too complex for a test like this, and that complexity doesn't actually relate directly to what is being tested here. In the meantime, perhaps you could just delete this function since you are not using it?